frontend/src/v5/ui/components/shared/appBar/appBar.component.tsx

```diff
@@ -15,31 +15,153 @@
+import { useParams } from 'react-router';
+import { useRouteMatch } from 'react-router-dom';
+import { AppBar as MuiAppBar, ClickAwayListener, Grow } from '@material-ui/core';
 
 import LogoIcon from '@assets/icons/logo.svg';
 import IntercomIcon from '@assets/icons/intercom.svg';
 import NotificationsIcon from '@assets/icons/notifications.svg';
 import { CircleButton } from '@/v5/ui/controls/circleButton';
-import { AvatarButton } from '@/v5/ui/controls/avatarButton';
-import { Items } from './appBar.styles';
+import ContactUsIcon from '@assets/icons/email.svg';
+import InviteAFriendIcon from '@assets/icons/add_user.svg';
+import TeamspacesIcon from '@assets/icons/teamspaces.svg';
+import VisualSettingsIcon from '@assets/icons/settings.svg';
+import SupportCentreIcon from '@assets/icons/question_mark.svg';
+import { CurrentUserHooksSelectors } from '@/v5/services/selectorsHooks/currentUserSelectors.hooks';
+import { AvatarButton } from '@controls/avatarButton';
+import {
+	Items,
+	UserMenu,
+	Section,
+	AvatarSection,
+	Popper,
+	Paper,
+	UserFullName,
+	UserUserName,
+	SignOutButton,
+	EditProfileButton,
+	Avatar,
+} from './appBar.styles';
 import { Breadcrumbs } from '../breadcrumbs';
+import { UserMenuButton } from './userMenuButton/userMenuButton.component';
 
-export const AppBar = (): JSX.Element => (
-	<MuiAppBar position="static">
-		<Items>
-			<LogoIcon />
-			<Breadcrumbs />
-		</Items>
-		<Items>
-			<CircleButton variant="contrast" aria-label="intercom">
-				<IntercomIcon />
-			</CircleButton>
-			<CircleButton variant="contrast" aria-label="notifications">
-				<NotificationsIcon />
-			</CircleButton>
-			<AvatarButton> GH </AvatarButton>
-		</Items>
-	</MuiAppBar>
+const getUserNamesInitials = ({ firstName, lastName }) => (
+	[firstName, lastName]
+		.map((name) => name.trim().charAt(0).toUpperCase())
+		.join('')
 );
+
+export const AppBar = (): JSX.Element => {
+	const { teamspace } = useParams();
+	const { url } = useRouteMatch();
+	const baseUrl = url.split('/').slice(0, 3).join('/');
+	const userIsPending = CurrentUserHooksSelectors.selectIsPending();
+	const user = CurrentUserHooksSelectors.selectCurrentUser();
+
+	const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+	const handleCloseDropdown = () => {
+		setAnchorEl(null);
+	};
+
+	const handleClickDropdown = (event: MouseEvent<HTMLButtonElement>) => {
+		event.stopPropagation();
+		setAnchorEl(event.currentTarget);
+	};
+
+	return (
+		<>
+			<MuiAppBar position="static">
+				<Items>
+					<LogoIcon />
+					<Breadcrumbs />
+				</Items>
+				<Items>
+					<CircleButton variant="contrast" aria-label="intercom">
+						<IntercomIcon />
+					</CircleButton>
+					<CircleButton variant="contrast" aria-label="notifications">
+						<NotificationsIcon />
+					</CircleButton>
+					<AvatarButton
+						onClick={handleClickDropdown}
+						src={user.hasAvatar ? user.avatarUrl : null}
+					>
+						{!userIsPending && getUserNamesInitials(user)}
```

Comment:
We can modify the original `<AvatarButton/>` component  and have a user as a prop,  so everytime we use AvatarButton it would know how to display the avatar image or initials.

Reply:
Fixed. I also renamed `AvatarButton` to just `Avatar` and it automatically detects whether the avatar is a button (so the cursor becomes a pointer).